src/components/mapImpl.js

```diff
@@ -124,6 +126,13 @@ export default {
     }
   },
 
+  beforeDestroy () {
+    const recycleKey = this.options.recycle ? recyclePrefix + this.options.recycle : recyclePrefix
```

Comment:
I not understand why you concat a `string` with a `boolean`?, on the other hand whats is the differences between truthy state and falsy state? Can you move this code to a function? you repeat that on mounted hook.

Reply:
It's supposed to be used as a different cache key, as suggested by @xkjyeah, so a user could supply option { recycle: 'keep-alive-key' }

But if you just recycle the same map everywhere, you can use boolean as well.

Reply:
What happend if the user set `{recycle: false}` the cache doesn't disable it still be cached, is this the behavior that you expect?

Reply:
Ok, I miss this part of the code
```js
      if (this.options.recycle && window[recycleKey]) {
        element.appendChild(window[recycleKey].div)
        this.$mapObject = window[recycleKey].map
        this.$mapObject.setOptions(options)
      }
```
I think that this is ok and redy to merge! the only requested change is move this validation
```javascript
setRecycleKey(options) {
    return options.recycle ? `${recyclePrefix}${options.recycle}` : recyclePrefix
}

// instead of write the same validation on differents hooks
// call the function in both places

const recycleKey = setRecycleKey(this.options)
```